### PR TITLE
Tidy up workaround fixes

### PR DESCRIPTION
**Pull request by trizin**
- New function `get_pair_timeframe_source_from_contract`
- Tests

### CODE REVIEW

**Comment by calina-c** [comment]
LGTM. I'm not approving since I was not requested, but I figured I'd give this a look

File: pdr_backend/subgraph/subgraph_feed_contracts.py [hunk 1]
```diff
@@ -75,11 +76,9 @@ def query_feed_contracts(
...
                 break
             for contract in contract_list:
-                pair = contract["token"]["name"]
-                timeframe = "5m" if int(contract["secondsPerEpoch"]) == 300 else "1h"
-                source = "binance"  # fix me
-                if None in (pair, timeframe, source):
-                    continue
+                pair, timeframe, source = get_pair_timeframe_source_from_contract(
+                    contract
+                )
```
Comment (idiom-bytes) [inline pdr_backend/subgraph/subgraph_feed_contracts.py:81]: nice1

File: pdr_backend/subgraph/info725.py [hunk 2]
```diff
@@ -90,3 +92,21 @@ def info725_to_info(info725: list) -> Dict[str, Optional[str]]:
...
+def get_pair_timeframe_source_from_contract(contract):
+    if contract["token"]["nft"]:
+        info725 = contract["token"]["nft"]["nftData"]
+        info = info725_to_info(info725)  # {"pair": "ETH/USDT", }
+        pair = info["pair"]
+        timeframe = info["timeframe"]
+        source = info["source"]
+        return (pair, timeframe, source)
+
+    if contract["id"] in WHITELIST_FEEDS_MAINNET:
```
Comment (idiom-bytes) [inline pdr_backend/subgraph/info725.py:106]: i have a feeling this will be the result 99% of the time, but i get why you made this choice

**Review by idiom-bytes — APPROVED** [review]
Looks good, thanks for moving this into a common fn

**Pull request merged**